agent/structs/dns.go

```diff
@@ -1,19 +1,9 @@
 // Copyright (c) HashiCorp, Inc.
 // SPDX-License-Identifier: BUSL-1.1
 
-package dns
+package structs
 
-import (
-	"math/rand"
-	"regexp"
-)
-
-// MaxLabelLength is the maximum length for a name that can be used in DNS.
-const MaxLabelLength = 63
-
-// InvalidNameRe is a regex that matches characters which can not be included in
-// a DNS name.
-var InvalidNameRe = regexp.MustCompile(`[^A-Za-z0-9\\-]+`)
```

Comment:
I very much didn't like moving these around to deal with the circular dependency, but I feel like this was the least gross option:
1. (this PR) Move conflicting `agent/dns` fields into lower-level packages 
2. Make sure `agent/dns` doesn't import `agent/config`, which means the agent package will need to translate runtime config into the DNS-specific, reloadable config.
3. Leave everything in `agent`, then have a bunch of files like `dns_v2_blah`